### PR TITLE
First iteration of the user settings and logout functionalities

### PR DESCRIPTION
**Pull request by CalexisGonzalez**
User settings activity implementation with modifiable fields, and log out functionality

### CODE REVIEW

File: app/src/main/java/com/example/fittracker/activity/UserSettingsActivity.java [hunk 1]
```diff
@@ -26,38 +25,41 @@ protected void onCreate(Bundle savedInstanceState) {
...
-        presenter = new UserSettingsPresenter(new UserSettingsView(
-                this),new UserSettingsModel(UserRoomDatabase.getDatabase(this)));
-    }
+    public void init() {
+        presenter = new UserSettingsPresenter(new UserSettingsView(this)
+                ,new UserSettingsModel(UserRoomDatabase.getDatabase(this))
+                ,getSharedPreferences(ConstantUtils.USER_PREFERENCES, MODE_PRIVATE)
+                .getInt(ConstantUtils.USER_PREFERENCES_ID,ConstantUtils.ZERO));
+        presenter.onCreate();
+    }//Check if its okay to pass the shared pref via constructor to presenter
```
Comment (SartiMau) [inline app/src/main/java/com/example/fittracker/activity/UserSettingsActivity.java]: No, the sharedPreferences must be in the model

File: app/src/main/java/com/example/fittracker/mvp/view/MainScreenView.java [hunk 2]
```diff
@@ -23,4 +26,21 @@ public void onSettingsPressed() {
...
+        startMain.addCategory(Intent.CATEGORY_HOME);
+        startMain.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+        activity.get().startActivity(startMain);
+    }
+
+    @Override
+    public void onLogOutPressed() {
+        SharedPreferences.Editor editor = activity.get().getSharedPreferences(ConstantUtils.USER_PREFERENCES, Context.MODE_PRIVATE)
+                .edit();
+        editor.putInt(ConstantUtils.USER_PREFERENCES_ID,-1);
```
Comment (SartiMau) [inline app/src/main/java/com/example/fittracker/mvp/view/MainScreenView.java]: use constants pls

File: app/src/main/java/com/example/fittracker/activity/UserSettingsActivity.java [hunk 3]
```diff
@@ -26,38 +25,41 @@ protected void onCreate(Bundle savedInstanceState) {
...
-    public void init(){
-        presenter = new UserSettingsPresenter(new UserSettingsView(
-                this),new UserSettingsModel(UserRoomDatabase.getDatabase(this)));
-    }
+    public void init() {
+        presenter = new UserSettingsPresenter(new UserSettingsView(this)
+                ,new UserSettingsModel(UserRoomDatabase.getDatabase(this))
+                ,getSharedPreferences(ConstantUtils.USER_PREFERENCES, MODE_PRIVATE)
+                .getInt(ConstantUtils.USER_PREFERENCES_ID,ConstantUtils.ZERO));
+        presenter.onCreate();
```
Comment (SartiMau) [inline app/src/main/java/com/example/fittracker/activity/UserSettingsActivity.java]: change the name of this method, call it init()

File: app/src/main/java/com/example/fittracker/mvp/presenter/UserSettingsPresenter.java [hunk 4]
```diff
@@ -1,47 +1,60 @@
...
+        view.onCancelClick();
     }
 
     @Override
     public void onCreate() {
-        //TODO get user data from db,print it in the view
+        // TODO get user data from db,print it in the view
+        User user = model.getUserData(id);
+        if (user != null) {
+            view.onCreate(user.getMail(), user.getPassword(), user.getName(), user.getSurname());
```
Comment (SartiMau) [inline app/src/main/java/com/example/fittracker/mvp/presenter/UserSettingsPresenter.java]: set each field value separately

File: app/src/main/java/com/example/fittracker/mvp/view/UserSettingsView.java [hunk 5]
```diff
@@ -1,8 +1,10 @@
 package com.example.fittracker.mvp.view;
```
Comment (SartiMau) [inline app/src/main/java/com/example/fittracker/mvp/view/UserSettingsView.java:1]: Why do you have a TextView and an EditText for each field?  Use only an EditText

File: app/src/main/java/com/example/fittracker/mvp/view/UserSettingsView.java [hunk 6]
```diff
@@ -34,33 +36,66 @@ public int hashCode() {
     }
 
     @Override
-    public void onEmailFocusChange() {
-
+    public void onEmailClick() {
+        mailView.setVisibility(View.GONE);
+        mailEdit.setVisibility(View.VISIBLE);
+        resetName();
+        resetPassword();
```
Comment (SartiMau) [inline app/src/main/java/com/example/fittracker/mvp/view/UserSettingsView.java]: why when you click on the email field you set the visibility of the password?

if this is necessary this must be in the presenter

File: app/src/main/java/com/example/fittracker/mvp/view/UserSettingsView.java [hunk 7]
```diff
@@ -34,33 +36,66 @@ public int hashCode() {
...
+        resetSurname();
```
Comment (SartiMau) [inline app/src/main/java/com/example/fittracker/mvp/view/UserSettingsView.java]: same

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/fittracker/mvp/view/MainScreenView.java (hunk 2 above)
Comment (CalexisGonzalez) [reply app/src/main/java/com/example/fittracker/mvp/view/MainScreenView.java]: Done

File: app/src/main/java/com/example/fittracker/activity/UserSettingsActivity.java (hunk 3 above)
Comment (CalexisGonzalez) [reply app/src/main/java/com/example/fittracker/activity/UserSettingsActivity.java]: Done

File: app/src/main/java/com/example/fittracker/mvp/presenter/UserSettingsPresenter.java (hunk 4 above)
Comment (CalexisGonzalez) [reply app/src/main/java/com/example/fittracker/mvp/presenter/UserSettingsPresenter.java]: Done

File: app/src/main/java/com/example/fittracker/activity/UserSettingsActivity.java (hunk 1 above)
Comment (CalexisGonzalez) [reply app/src/main/java/com/example/fittracker/activity/UserSettingsActivity.java]: Done

File: app/src/main/java/com/example/fittracker/mvp/view/UserSettingsView.java (hunk 6 above)
Comment (CalexisGonzalez) [reply app/src/main/java/com/example/fittracker/mvp/view/UserSettingsView.java]: Done

File: app/src/main/java/com/example/fittracker/mvp/view/UserSettingsView.java (hunk 7 above)
Comment (CalexisGonzalez) [reply app/src/main/java/com/example/fittracker/mvp/view/UserSettingsView.java]: Done

**Comment by CalexisGonzalez** [comment]
All the intent and mail related fields will be used when the whole automatic mail systems is implemented
(is not currently being used)

File: app/src/main/java/com/example/fittracker/activity/UserSettingsActivity.java [hunk 8]
```diff
@@ -27,28 +27,27 @@ protected void onCreate(Bundle savedInstanceState) {
...
-        presenter.onCreate();
-    }//Check if its okay to pass the shared pref via constructor to presenter
+                ,new UserSettingsModel(UserRoomDatabase.getDatabase(this)
+                ,getSharedPreferences(ConstantUtils.USER_PREFERENCES, MODE_PRIVATE)));
+        presenter.init();
+    }
 
-    @OnClick(R.id.usersettings_activity_textview_email)
+    @OnClick(R.id.usersettings_activity_edittext_email)
     public void onEmailFocusChange() {
```
Comment (SartiMau) [inline app/src/main/java/com/example/fittracker/activity/UserSettingsActivity.java]: if you need it.. you can put a boolean parameter so if you are focused will be true, and false if you are leaving the field.

public void onEmailFocusChange(boolean isFocused)

**Review by SartiMau — APPROVED** [review]

File: app/src/main/java/com/example/fittracker/activity/UserSettingsActivity.java (hunk 8 above)
Comment (CalexisGonzalez) [reply app/src/main/java/com/example/fittracker/activity/UserSettingsActivity.java]: All the focuschange views got replaced by normal onclick views.

File: app/src/main/java/com/example/fittracker/GMailSender.java [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+
+    public String getSenderUser(){return user;}
+
+    public synchronized void sendMail(String subject, String body, String sender, String recipients) throws Exception {
+        MimeMessage message = new MimeMessage(session);
+        DataHandler handler = new DataHandler(new ByteArrayDataSource(body.getBytes(), ConstantUtils.DATAHANDLER_TYPE));
+        message.setSender(new InternetAddress(sender));
+        message.setSubject(subject);
+        message.setDataHandler(handler);
+        if (recipients.indexOf(',') > 0)
```
Comment (lucianoeli) [inline app/src/main/java/com/example/fittracker/GMailSender.java]: same

**Review by lucianoeli — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/fittracker/GMailSender.java (hunk 9 above)
Comment (CalexisGonzalez) [reply app/src/main/java/com/example/fittracker/GMailSender.java]: Done

File: app/src/main/java/com/example/fittracker/DbQuery/DbFetchUserData.java [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+    @Override
+    protected User doInBackground(Integer... ids) {
+        return (userDao.fetchUserData(ids[ConstantUtils.ZERO]));
+    }
+
+    @Override
+    public User executeQuery(Integer id) {
+        try {
+            return this.execute(id).get();
+        } catch (ExecutionException e) {
```
Comment (priettt) [inline app/src/main/java/com/example/fittracker/DbQuery/DbFetchUserData.java]: You could catch multiple exceptions in a line with the "|" operator.

catch (ExecutionException | InterruptedException e)

File: app/src/main/java/com/example/fittracker/GMailSender.java [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+import javax.activation.DataSource;
+import javax.mail.Message;
+import javax.mail.PasswordAuthentication;
+import javax.mail.Session;
+import javax.mail.Transport;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeMessage;
+
+public class GMailSender extends javax.mail.Authenticator {
+    private String mailhost = "smtp.gmail.com";
```
Comment (priettt) [inline app/src/main/java/com/example/fittracker/GMailSender.java]: If mailhost is a constant, make it static final and name it in UPPER_CASE

File: app/src/main/java/com/example/fittracker/DbQuery/DbFetchUserData.java (hunk 10 above)
Comment (CalexisGonzalez) [reply app/src/main/java/com/example/fittracker/DbQuery/DbFetchUserData.java]: Done

File: app/src/main/java/com/example/fittracker/GMailSender.java (hunk 11 above)
Comment (CalexisGonzalez) [reply app/src/main/java/com/example/fittracker/GMailSender.java]: Done

**Review by lucianoeli — APPROVED** [review]

**Review by SartiMau — APPROVED** [review]

File: app/src/main/java/com/example/fittracker/DbQuery/DbFetchUserId.java [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+    @Override
+    protected Integer doInBackground(User... users) {
+        return userDao.fetchUserId(users[ConstantUtils.ZERO].getMail());
+    }
+
+    @Override
+    public Integer executeQuery(User user) {
+        try {
+            return this.execute(user).get();
+        } catch (ExecutionException e) {
```
Comment (priettt) [inline app/src/main/java/com/example/fittracker/DbQuery/DbFetchUserId.java]: Same here

File: app/src/main/java/com/example/fittracker/DbQuery/DbQueryUserValid.java [hunk 13]
```diff
@@ -23,7 +23,7 @@ protected Boolean doInBackground(User... users) {
     @Override
     public Boolean executeQuery(User user) {
         try {
-            return new DbQueryUserValid(userDao).execute(user).get();
+            return this.execute(user).get();
         } catch (ExecutionException e) {
```
Comment (priettt) [inline app/src/main/java/com/example/fittracker/DbQuery/DbQueryUserValid.java]: same here

File: app/src/main/java/com/example/fittracker/DbQuery/DbFetchUserId.java (hunk 12 above)
Comment (CalexisGonzalez) [reply app/src/main/java/com/example/fittracker/DbQuery/DbFetchUserId.java]: Done

File: app/src/main/java/com/example/fittracker/DbQuery/DbQueryUserValid.java (hunk 13 above)
Comment (CalexisGonzalez) [reply app/src/main/java/com/example/fittracker/DbQuery/DbQueryUserValid.java]: Done

File: app/src/main/java/com/example/fittracker/activity/UserSettingsActivity.java [hunk 14]
```diff
@@ -27,8 +27,8 @@ protected void onCreate(Bundle savedInstanceState) {
 
     public void init() {
         presenter = new UserSettingsPresenter(new UserSettingsView(this)
-                ,new UserSettingsModel(UserRoomDatabase.getDatabase(this)
-                ,getSharedPreferences(ConstantUtils.USER_PREFERENCES, MODE_PRIVATE)));
+                , new UserSettingsModel(UserRoomDatabase.getDatabase(this)
```
Comment (priettt) [inline app/src/main/java/com/example/fittracker/activity/UserSettingsActivity.java]: When breaking lines in method calls, leave the commas at the end.

**Pull request merged**